crates/turborepo-lib/src/task_hash.rs

```diff
@@ -448,22 +449,24 @@ impl<'a> TaskHasher<'a> {
     pub fn env(
         &self,
         task_id: &TaskId,
-        task_env_mode: ResolvedEnvMode,
+        task_env_mode: EnvMode,
         task_definition: &TaskDefinition,
         global_env: &EnvironmentVariableMap,
     ) -> Result<EnvironmentVariableMap, Error> {
         match task_env_mode {
-            ResolvedEnvMode::Strict => {
+            EnvMode::Strict => {
                 let mut pass_through_env = EnvironmentVariableMap::default();
                 let default_env_var_pass_through_map =
                     self.env_at_execution_start.from_wildcards(&[
                         "SHELL",
                         // Command Prompt casing of env variables
+                        "APPDATA",
```

Comment:
This is necessary due to older versions of `pnpm` not having https://github.com/pnpm/npm-conf/pull/10 which results in `pnpm` crashing if `APPDATA` isn't defined.

Technically we could have users define this instead of baking it into `turbo` itself, but considering the popularity of `pnpm` and the amount of digging it took to find that PR, it seems nice to include it.